client/state/sites/selectors.js

```diff
@@ -104,17 +105,21 @@ export function isRequestingSites( state ) {
 }
 
 /**
- * Returns true if an url is already in the current site list
- * @param {Object}	state Global state tree
- * @return {Boolean}
+ * Returns a site object by its URL.
+ *
+ * @param  {Object}  state Global state tree
+ * @param  {String}  url   Site URL
+ * @return {?Object}       Site object
  */
 export function getSiteByUrl( state, url ) {
-	for ( let siteId in state.sites.items ) {
-		const siteSlug = getSiteSlug( state, siteId );
-		const urlSlug = url.replace( /^https?:\/\//, '' ).replace( /\//g, '::' );
-		if ( siteSlug === urlSlug ) {
-			return state.sites.items[ siteId ];
-		}
+	const slug = url.replace( /^https?:\/\//, '' ).replace( /\//g, '::' );
```

Comment:
As bare as this is, I think we need to convert it into a utility. There's quite a few places where we could use this, and it'd be nice to make sure it's always consistent.